lib/sdf-server/src/server/service/v2/module/sync.rs

```diff
@@ -30,119 +26,24 @@ pub async fn sync(
-
-    track(
-        &posthog_client,
-        &ctx,
-        &original_uri,
-        "sync",
-        serde_json::json!({}),
-    );
-
-    Ok(Json(synced_modules))
-}
-
-#[instrument(
-    name = "sdf.v2.module.sync.sync_inner"
-    skip_all,
-    level = "debug",
-)]
-async fn sync_inner(
-    ctx: &DalContext,
-    raw_access_token: &str,
-) -> Result<frontend_types::SyncedModules, ModulesAPIError> {
-    let start = Instant::now();
-
-    // Collect all user facing schema variants. We need to see what can be upgraded.
-    let schema_variants = SchemaVariant::list_user_facing(ctx).await?;
-
-    // Find all local hashes and mark all seen schemas.
-    let mut local_hashes = HashMap::new();
-    let mut seen_schema_ids = HashSet::new();
-    for schema_variant in &schema_variants {
-        let schema_id: SchemaId = schema_variant.schema_id.into();
-        seen_schema_ids.insert(schema_id);
-
-        if let Entry::Vacant(entry) = local_hashes.entry(schema_id) {
-            let local_module = Module::find_for_member_id(ctx, schema_id)
-                .await?
-                .ok_or(ModulesAPIError::ModuleNotFoundForSchema(schema_id))?;
-            entry.insert(local_module.root_hash().to_owned());
-        }
-    }
-
-    // Collect the latest modules.
     let latest_modules = {
         let module_index_url = ctx
             .module_index_url()
             .ok_or(ModulesAPIError::ModuleIndexNotConfigured)?;
         let module_index_client =
-            ModuleIndexClient::new(module_index_url.try_into()?, raw_access_token);
+            ModuleIndexClient::new(module_index_url.try_into()?, &raw_access_token);
         module_index_client.list_latest_modules().await?
     };
 
-    // Begin populating synced modules.
-    let mut synced_modules = frontend_types::SyncedModules::new();
-
-    // Group the latest hashes by schema. Populate installable modules along the way.
-    let mut latest_modules_by_schema: HashMap<SchemaId, frontend_types::LatestModule> =
-        HashMap::new();
-    for latest_module in latest_modules.modules {
-        let schema_id: SchemaId = latest_module
-            .schema_id()
-            .ok_or(ModulesAPIError::ModuleMissingSchemaId(
-                latest_module.id.to_owned(),
-                latest_module.latest_hash.to_owned(),
-            ))?
-            .into();
-        match latest_modules_by_schema.entry(schema_id) {
-            Entry::Occupied(entry) => {
-                let existing: frontend_types::LatestModule = entry.get().to_owned();
-                return Err(ModulesAPIError::LatestModuleTooManyForSchema(
-                    schema_id,
-                    existing.latest_hash,
-                    latest_module.latest_hash,
-                ));
-            }
-            Entry::Vacant(entry) => {
-                entry.insert(latest_module.to_owned());
-            }
-        }
+    let synced_modules = Module::sync(&ctx, latest_modules.modules).await?;
```

Comment:
Now, we collect the `latest_modules` first, drop our client out of scope and then `sync`.